mhs/outbound/outbound/request/synchronous/handler.py

```diff
@@ -111,6 +108,57 @@ def _extract_interaction_id(self):
                                         reason='Required Interaction-Id header not found') from e
         return interaction_id
 
+    def _extract_default_workflow(self, interaction_details, interaction_id):
+        try:
+            wf = self.workflows[interaction_details['workflow']]
+        except KeyError as e:
+            logger.error('0008', "Wasn't able to determine workflow for {InteractionId} . This likely is due to a "
+                                 "misconfiguration in interactions.json", {"InteractionId": interaction_id})
+            raise tornado.web.HTTPError(500,
+                                        f"Couldn't determine workflow to invoke for interaction ID: {interaction_id}",
+                                        reason=f"Couldn't determine workflow to invoke for interaction ID: "
+                                        f"{interaction_id}") from e
+        return wf
+
+    def _extract_sync_async_from_interaction_details(self, interaction_details):
+        is_sync_async = interaction_details.get('sync_async')
+        if is_sync_async is None:
+            logger.error('0032', 'Failed to retrieve sync-async flag from interactions.json')
+            raise tornado.web.HTTPError(500, f'Failed to parse sync-async flag from interactions.json file',
+                                        reason='Failed to parse sync-async flag from interactions file for this '
+                                               'interaction, contact your administrator')
+        return is_sync_async
+
+    def _should_invoke_sync_async_workflow(self, interaction_config, sync_async_header):
+        if interaction_config and sync_async_header:
+            return True
+        elif (not interaction_config) and sync_async_header:
+            logger.error('0033', 'Message header requested sync-async wrap for un-supported sync-async')
+            raise tornado.web.HTTPError(400, f'Message header requested sync-async wrap for un-supported sync-async',
+                                        reason='Message header requested sync-async wrap for a message pattern'
+                                               'that does not support sync-async')
+        elif interaction_config and (not sync_async_header):
+            return False
+        else:
+            return False
```

Comment:
Looking at the code coverage report, this line isn't hit. Also, the last 2 cases are the same and can be combined.